src/components/NSLookupTool/NSLookupTool.tsx

```diff
@@ -60,70 +64,65 @@ export function NSLookup() {
     );
-    /**
-     * onSubmit: Handler function that is triggered when the form is submitted.
-     * It prepares the arguments and initiates the execution of the `bed` command.
-     * Upon successful execution, it updates the state with the process PID and output.
-     * If an error occurs during the command execution, it updates the output with the error message.
-     * @param {FormValues} values - An object containing the form input values.
-     */
-
-    // Actions taken after saving the output
+
+    // Updating the state of Saving the output
+
     const handleSaveComplete = () => {
-        // Indicating that the file has saved which is passed
-        // back into SaveOutputToTextFile to inform the user
-        setHasSaved(true);
-        setAllowSave(false);
+        setHasSaved(true); // Mark output as saved
+        setAllowSave(false); // Disabling the Save option
     };
 
-    const onSubmit = (values: FormValues) => {
-        // Disallow saving until the tool's execution is complete
-        setAllowSave(false);
+    // Handling the submitted form
 
-        setLoading(true);
-        const args = ["-s", values.ipaddress];
-        CommandHelper.runCommandGetPidAndOutput("nslookup", args, handleProcessData, handleProcessTermination)
+    const onSubmit = (values: FormValuesType) => {
+        const ipAddress = values.ipAddress.trim(); // Removing the extra white space
+        if (!ipAddress) {
+            setOutput("Error: No IP address or hostname provided."); // Error Message
+            return;
+        }
+
+        setAllowSave(false); // Disabling the save option before start
+        setLoading(true); // Show loading indication
+
+        CommandHelper.runCommandGetPidAndOutput("nslookup", [ipAddress], handleProcessData, handleProcessTermination)
             .then(({ pid, output }) => {
-                setPid(pid);
-                setOutput(output);
+                setPid(pid); // Setting the Process ID
+                setOutput(output); // Setting the command output
             })
             .catch((error) => {
-                setLoading(false);
-                setOutput(`Error: ${error.message}`);
+                setLoading(false); //Stop Loading
+                setOutput(`Error: ${error.message}`); // Display the error message
             });
     };
 
-    /**
-     * clearOutput: A callback function to clear the current output displayed in the UI.
-     * This function resets the output state to an empty string.
-     */
+    // Clearing the output
+
     const clearOutput = useCallback(() => {
-        setOutput("");
-        setHasSaved(false);
-        setAllowSave(false);
-    }, [setOutput]); // Dependency on the setOutput function.
+        setOutput(""); // Clears output
+        setHasSaved(false); // Reset save status
+        setAllowSave(false); // Disable saving
+    }, [setOutput]);
 
     return (
         <form onSubmit={form.onSubmit((values) => onSubmit(values))}>
```

Comment:
Can add RenderComponent and loadingModal here. 
```suggestion
  <>
            <RenderComponent
                title={title}
                description={description}
                steps={steps}
                tutorial={tutorial}
                sourceLink={sourceLink}
            >
            {!loadingModal && (
                    <InstallationModal
                        isOpen={opened}
                        setOpened={setOpened}
                        feature_description={description}
                        dependencies={dependencies}
                    ></InstallationModal>
                )}
        <form onSubmit={form.onSubmit((values) => onSubmit(values))}>
```